angularjs-portal-home/src/main/webapp/partials/widgets/weather.html

```diff
@@ -25,6 +29,44 @@
           </a>
         </li>
     </ul>
+    
+    <div ng-if="weather.location == null">
```

Comment:
suggest using the falsy check instead. `ng-if="!weather.location"`. That will check for null and undefined.